src/sentry/static/sentry/app/views/settings/organizationFiltersAndSampling/rules/rule/projects.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+
+import PlatformList from 'app/components/platformList';
+import {t} from 'app/locale';
+import {Organization, Project} from 'app/types';
+import Projects from 'app/utils/projects';
+import withOrganization from 'app/utils/withOrganization';
+
+type Props = {
+  projectIds: Array<number>;
+  organization: Organization;
+};
+
+function ProjectList({projectIds, organization}: Props) {
+  if (!projectIds.length) {
+    return <React.Fragment>{t('All')}</React.Fragment>;
+  }
+
+  return (
+    <Projects orgId={organization.slug} allProjects>
```

Comment:
Can you pass here only project ids you need? `allProjects` prop might be a bit too much especially with larger orgs (some of them have 1000+ projects)

Reply:
Matej I agree with your suggestion, but Projects only accepts a list of slugs and that is not what I get.
I thought about using `withProject`, which checks the Project store, but you mentioned that it's not ideal

Reply:
ok I have added one more check

Reply:
@priscilawebdev you should talk with @davidenwang as he was having a similar problem.

Reply:
had a look, seems like I have the opposite problem since I'm trying to convert slugs -> ids so I can pass in the list of project slugs. I'm wondering if it's worth it to add an option to pass id's to the `Projects` component not sure if that would mess up the typing a lot though...

Reply:
@RaduW will check if we can have project slugs instead of project ids 😉